src/Ydb.Sdk/src/Driver.cs

```diff
@@ -72,6 +73,17 @@ public ValueTask DisposeAsync()
 
     public async Task Initialize()
     {
+        if (_config.Credentials is IUseDriverConfig useDriverConfig)
+        {
+            await useDriverConfig.ProvideConfig(_config);
+            if (_config.Credentials is StaticCredentialsProvider staticCredentialsProvider)
```

Comment:
Don't check class of the credentials.

Add Initialize method to IcredentialsProvider interface with noop default implementation and call the method always.

It allow initialize any provider if need, not StaticCredentialsProvider only.